src/reynir/bintokenizer.py

```diff
@@ -1047,7 +1060,11 @@ def parse_phrases_2(
         at_sentence_start = False
 
         while True:
-            next_token = next(token_stream)
+            next_token = (
+                following_tokens.popleft()
+                if len(following_tokens)
```

Comment:
Interesting solution! One question: is `len(following_tokens)` needed or can you do just `if following_tokens`?

Reply:
Good point! len() isn't needed.